app/screens/custom_status/custom_status_modal.tsx

```diff
@@ -0,0 +1,391 @@
+                            emojiName={emoji || 'speech_balloon'}
+                            size={20}
+                        />
+                    </Text>
+                ) : (
+                    <CompassIcon
+                        name='emoticon-happy-outline'
+                        size={24}
+                        style={style.icon}
+                    />
+                )}
+            </TouchableOpacity>
+        );
+
+        const customStatusInput = (
+            <View style={style.inputContainer}>
+                <TextInput
+                    testID='custom_status.input'
+                    autoCapitalize='none'
+                    autoCorrect={false}
+                    blurOnSubmit={false}
+                    disableFullscreenUI={true}
+                    keyboardAppearance={getKeyboardAppearanceFromTheme(theme)}
+                    keyboardType='default'
+                    maxLength={CustomStatus.CUSTOM_STATUS_TEXT_CHARACTER_LIMIT}
+                    onChangeText={this.handleTextChange}
+                    placeholder={this.props.intl.formatMessage({id: 'custom_status.set_status', defaultMessage: 'Set a Status'})}
+                    placeholderTextColor={changeOpacity(theme.centerChannelColor, 0.5)}
+                    returnKeyType='go'
+                    style={style.input}
+                    secureTextEntry={false}
+                    underlineColorAndroid='transparent'
+                    value={text}
+                />
+                {customStatusEmoji}
+                {isStatusSet ? (
+                    <View
+                        style={style.clearButton}
+                        testID='custom_status.input.clear.button'
+                    >
+                        <ClearButton
+                            handlePress={this.clearHandle}
+                            theme={theme}
+                        />
+                    </View>
+                ) : null}
+            </View>
+        );
+
+        let keyboardOffset = DeviceTypes.IS_IPHONE_WITH_INSETS ? 110 : 60;
+        if (isLandscape) {
+            keyboardOffset = DeviceTypes.IS_IPHONE_WITH_INSETS ? 0 : 10;
+        }
+
+        return (
+            <SafeAreaView
+                testID='custom_status.screen'
+                style={style.container}
+            >
+                <KeyboardAvoidingView
+                    behavior='padding'
+                    style={style.container}
+                    keyboardVerticalOffset={keyboardOffset}
+                    enabled={Platform.OS === 'ios'}
+                >
+                    <ScrollView
+                        bounces={false}
+                    >
+                        <StatusBar/>
+                        <View style={style.scrollView}>
+                            {customStatusInput}
+                            {this.renderRecentCustomStatuses()}
+                            {this.renderCustomStatusSuggestions()}
```

Comment:
perhaps you can pass `style` here and avoid calling `getStyleSheet(theme)` in each individual function as the style would be the same